src/test/java/org/broadinstitute/hellbender/engine/GatkToolIntegrationTest.java

```diff
@@ -64,4 +67,25 @@ public void testBrokenReferenceDictionaryErrorMessage() throws IOException {
 
         runCommandLine(Arrays.asList(args), Mutect2.class.getSimpleName());
     }
+
+    @Test
+    public void testSharding() {
+        final String outDir = createTempDir("GTShardedOutput").getAbsolutePath();
+        final String fileBase = "test";
+        final String out = Paths.get(outDir, fileBase + FileExtensions.COMPRESSED_VCF).toString();
+        final String[] args = new String[] {
+                "-V",  TEST_DIRECTORY + "example_variants_withSequenceDict.vcf",
+                "-R", hg19MiniReference,
+                "--" + StandardArgumentDefinitions.MAX_VARIANTS_PER_SHARD_LONG_NAME, "10",
+                "-O", out};
+        runCommandLine(Arrays.asList(args), SelectVariants.class.getSimpleName());
+
+        // 11 total records in the test input should create 2 vcf shards with 10 and 1 records
+        final String firstShard = Paths.get(outDir, fileBase + ShardingVCFWriter.SHARD_INDEX_PREFIX + "0" + ShardingVCFWriter.SHARD_INDEX_SUFFIX).toString();
+        final String secondShard = Paths.get(outDir, fileBase + ShardingVCFWriter.SHARD_INDEX_PREFIX + "1" + ShardingVCFWriter.SHARD_INDEX_SUFFIX).toString();
+        final Pair<VCFHeader, List<VariantContext>> firstResults = VariantContextTestUtils.readEntireVCFIntoMemory(firstShard);
+        final Pair<VCFHeader, List<VariantContext>> secondResults = VariantContextTestUtils.readEntireVCFIntoMemory(secondShard);
+        Assert.assertEquals(firstResults.getValue().size(), 10, "First shard has wrong number of records");
+        Assert.assertEquals(secondResults.getValue().size(), 2, "Second shard has wrong number of records");
```

Comment:
Having experienced a lot of pain from lack of indexes, let's check for those too.

Reply:
Done